File: core/src/galenscovell/oregontrail/things/inanimate/Events/Planet.java
package galenscovell.oregontrail.things.inanimate.Events;

import java.util.Random;

public class Planet implements Event {
    private int resources, type, population;
    private int humidity, atmosphere, temperature, gravity;

    public Planet() {
        generate();
    }

    private void generate() {
        Random random = new Random();
    }

    @Override
    public void start() {

    }

    @Override
    public void end() {

    }

    @Override
    public void setDistanceTo(int distance) {

    }

    @Override
    public int getDistanceTo(int distance) {
        return 0;
    }
}

File: core/src/galenscovell/hinterstar/things/inanimate/Events/Planet.scala
package galenscovell.hinterstar.things.inanimate.Events

import java.util.Random


class Planet(distanceTo: Float) extends Event {
  private var resources: Int = 0
  private var planetType: Int = 0
  private var population: Int = 0
  private var humidity: Int = 0
  private var atmosphere: Int = 0
  private var temperature: Int = 0
  private var gravity: Int = 0

  private val distance: Float = distanceTo
  private val eventType: String = "Test"

  generate()


  private def generate(): Unit = {
    val random: Random = new Random
  }

  override def start(): Unit = {
    println("Planet event started")
  }

  override def end(): Unit = {
    println("Planet event ended")
  }

  override def getDistance: Float = {
    distance
  }

  override def getType: String = {
    eventType
  }
}
